lib/galaxy/model/base.py

```diff
@@ -52,11 +52,12 @@ def transaction(session: Union[scoped_session, Session, "SessionlessContext"]):
         yield
         return  # exit: can't use as a Session
 
-    if not session.in_transaction():  # type:ignore[union-attr]
-        with session.begin():  # type:ignore[union-attr]
-            yield
-    else:
-        yield
+    yield  # TODO once we can verify this does not break SQLAlchemy transactions, remmove this helper completely (561 instances)
+    # if not session.in_transaction():  # type:ignore[union-attr]
+    #    with session.begin():  # type:ignore[union-attr]
+    #        yield
+    # else:
+    #    yield
```

Comment:
Normally we rely on git history instead of leaving commented out code.

Reply:
I thought commenting out would be helpful in this case since (a) this is temporary (dev branch only: it should be removed or restored prior to 25.0); and (b) the function is meaningless without that code: the commented out section immediately provides context in case someone reads that code. But if you prefer this to be removed, I'll certainly do so.

Reply:
I think the helpful comment on line 55 already gives sufficient context to a reader, but obviously not a blocker!

Reply:
Done. I've edited the comment too.